### PR TITLE
Add unit testing and no build options

### PR DESCRIPTION
**Pull request by soundcloudfrank**
Adding switches to allow for using test command and disabling "build" command for use when executing test-while-building.

### CODE REVIEW

File: src/main/java/au/com/rayh/XCodeBuilder.java [hunk 1]
```diff
@@ -212,7 +218,7 @@ public XCodeBuilder(Boolean buildIpa, Boolean generateArchive, Boolean cleanBefo
     		String keychainName, String keychainPath, String keychainPwd, String symRoot, String xcodeWorkspaceFile,
     		String xcodeSchema, String buildDir, String developmentTeamName, String developmentTeamID, Boolean allowFailingBuildResults,
     		String ipaName, Boolean provideApplicationVersion, String ipaOutputDirectory, Boolean changeBundleID, String bundleID,
-    		String bundleIDInfoPlistPath, String ipaManifestPlistUrl, Boolean interpretTargetAsRegEx, String ipaExportMethod) {
+    		String bundleIDInfoPlistPath, String ipaManifestPlistUrl, Boolean interpretTargetAsRegEx, String ipaExportMethod, Boolean executeTests, Boolean skipBuild) {
```
Comment (aheritier) [inline src/main/java/au/com/rayh/XCodeBuilder.java]: For migration/upgrade purpose AFAIK you should create a copy of the existing construction with @Deprecated like it was done for others

Comment (soundcloudfrank) [inline src/main/java/au/com/rayh/XCodeBuilder.java]: Fixed.

Comment (soundcloudfrank) [inline src/main/java/au/com/rayh/XCodeBuilder.java]: Fixed.

**Comment by aheritier** [comment]
LGTM. Let's give some days to the community to review it cc @jenkinsci/code-reviewers

**Comment by aheritier** [comment]
New artifact available here to test : https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fxcode-plugin/detail/PR-81/3/artifacts

File: src/main/java/au/com/rayh/XCodeBuilder.java [hunk 2]
```diff
@@ -246,11 +252,32 @@ public XCodeBuilder(Boolean buildIpa, Boolean generateArchive, Boolean cleanBefo
...
 
     @Deprecated
     public XCodeBuilder(Boolean buildIpa, Boolean generateArchive, Boolean cleanBeforeBuild, Boolean cleanTestReports, String configuration,
                         String target, String sdk, String xcodeProjectPath, String xcodeProjectFile, String xcodebuildArguments,
+                        String cfBundleVersionValue, String cfBundleShortVersionStringValue, Boolean unlockKeychain,
+                        String keychainName, String keychainPath, String keychainPwd, String symRoot, String xcodeWorkspaceFile,
+                        String xcodeSchema, String buildDir, String developmentTeamName, String developmentTeamID, Boolean allowFailingBuildResults,
+                        String ipaName, Boolean provideApplicationVersion, String ipaOutputDirectory, Boolean changeBundleID, String bundleID,
+                        String bundleIDInfoPlistPath, String ipaManifestPlistUrl, Boolean interpretTargetAsRegEx, String ipaExportMethod) {
+        this(buildIpa, generateArchive, cleanBeforeBuild, cleanTestReports, configuration,
```
Comment (oleg-nenashev) [inline src/main/java/au/com/rayh/XCodeBuilder.java]: Well, this parameter set is rather a subject for redesign by somebody

**Comment by soundcloudfrank** [comment]
It'd be great to get some closure on this. Are there additional comments and/or would the maintainers like me to address the "test settings" in some way, or does this stand alone / is this ready to commit?

File: src/main/java/au/com/rayh/XCodeBuilder.java (hunk 2 above)
Comment (soundcloudfrank) [inline src/main/java/au/com/rayh/XCodeBuilder.java]: I'm not sure from your comment whether you still want changes made in this PR or not. Could you clarify?

Comment (aheritier) [inline src/main/java/au/com/rayh/XCodeBuilder.java]: @oleg-nenashev I'm not sure to fully understand the advice, but AFAIU it's more a generic advice to redesign this class ? It's not specific to this PR ? (I would love so much to have some experts proposing some cleanup for this plugin :-) )

**Pull request closed without merging**